### DISCUSSION TITLE
#[Url(history: true)] when updating multiple properties creates multiple (redundant) history steps

### DISCUSSION THREAD

**Discussion by goldmerc** [livewire/livewire#8636] [1. Help]
If I update two properties in the same request, both with #[Url(history: true)] then livewire creates two browser history entries rather than one.

For example - 

```
class TestQueryString extends Component
{
    #[Url(history: true)]
    public $minPrice;

    #[Url(history: true)]
    public $maxPrice;
}
```
```
<form wire:submit="$refresh">
    <input wire:model="minPrice"/>
    <input wire:model="maxPrice"/>
    <button type="submit">Update Price Range</button>
</form>
```
If I enter min price 20 and max price 50 and submit the form, I see the expected url - 

`app.test/price-range-test?minPrice=20&maxPrice=50`

However, when I hit back, I get - 

`app.test/price-range-test?minPrice=20`

But I would expect to go directly back to -

`app.test/price-range-test`

Is this the expected behaviour? Seems like a bug to me but maybe I'm missing something?

Is there a way to update multiple properties with #[Url(history: true)] without creating multiple history steps?

Thanks in advance,

Jonathan

**Comment by joshhanley** [comment]
@goldmerc Livewire controls the query string on the front end, so it's likely updating the URL each time an input is changed (that has URL). Hence why when you hit back, the previous change is after the first input change.

There is a URL replace option, but then I think it will just overwrite the URL completely and there won't be a back.

**Comment by joshhanley** [comment]
Closing in favour of #8640.
